charts/iter8lib/templates/_task-database.tpl

```diff
@@ -0,0 +1,17 @@
+{{- define "task.database" -}}
+- task: collect-metrics-database
+  with:
+    {{- if .Values.providers }}
+    providers:
+    {{- range .Values.providers }}
+      - {{ . }}
+    {{- end }}
+    {{- end }}
+    versionInfo:
+    - endpoint: {{ required "A valid endpoint value is required!" .Values.endpoint | toString }}
+      destination_workload: {{ required "A valid destination_workload value is required!" .Values.destination_workload | toString }}
+      destination_workload_namespace: {{ required "A valid destination_workload_namespace value is required!" .Values.destination_workload_namespace | toString }}
```

Comment:
We want to decouple Istio specific from this task. In other words, we can stuff everything in .Values.versionInfo and reference the versionInfo object here, instead of getting into specific info fields that are provider specific.